src/site/markdown/server/manage-webadmin.md

```diff
@@ -2496,6 +2497,45 @@ Response codes:
 
 Not implemented yet.
 
+## Deleted Messages Vault
+
+Deleted messages from an user will be stored in a restriction place where only delegated administrator can access.
+This place is called the Vault. Messages in the Vault will be deleted permanently by a configured retention policy.
+
+ - [Restore Deleted Messages](#Restore_deleted_messages)
+
+### Restore Deleted Messages
+
+An user can request to the administrator to restore his messages had been deleted and now at the Vault.
+To restore them, the administrator perform a http request:
+
+```
+curl -XPOST http://ip:port/deletedMessages/user/userToRestore@domain.ext
+```
+
+**All** messages from the user passed in the path param will be moved from the Vault to his new mailbox.
+Response code:
+
+ - 201: Task for restoring deleted has been created
```

Comment:
We are lacking the case of conflict: we probably never want two restore tasks to run for a given user at the same time, right?

Reply:
Should I implement to cover this case? or add a warning in the document is enough?

Reply:
IMO there is no need to have a `conflict` here. Duplication in the restore mailbox seems really OK to me.

Reply:
> IMO there is no need to have a `conflict` here. Duplication in the restore mailbox seems really OK to me.

The first thing that will happen is people sending duplicate REST queries (I know it because it's what happened for mailbox reindex when I worked with some sysadmins).

So it clearly makes sense to say : "hey, there's already a restore process in progress, look at this task".

BTW the task is not idempotent so there's chances the query is received twice because of network issues and the result will not be what's expected.

It's not an easy problem to solve but it's something we will need for most of our long running tasks.

Reply:
I can maintain a set in memory to keep users are in processing, but it doesn't work with distributed webadmin.

Reply:
> I can maintain a set in memory to keep users are in processing, but it doesn't work with distributed webadmin.

Please don't!

We need further discussions + careful planning for this.